TFLite/src/common/Storage.py

```diff
@@ -0,0 +1,208 @@
+import os
+from queue import Queue
+from google.cloud import spanner
+
+# Module to store Graph data into database
+# Node object attributes are stored in 'Operators' table
+# Edge object attributes are stored in 'Tensors' table
+# Graph object attributes are stored in 'Models' table
+
+# The attribute names in each case must correspond to a column name in 
+# the database. If any attribute is added in the Class files, 
+# the Storage class will automatically try to add that attribute to the 
+# database the next time it is run with any model
+
+# If an attribute is added to the class file and not to the database,
+# this will throw and error unless explicitly changed
+
+# Spanner allows only 20000 mutations per commit, so _load_operators
+# and _load_tensors implement batching
+class Storage:
+
+    # Initialize spanner client with instance and database
+    def __init__(self, instance_id, database_id):
+        self.spanner_client = spanner.Client()
+        self.instance = self.spanner_client.instance(instance_id)
+        self.database = self.instance.database(database_id)
+
+    # Internal function to load graph metadata into the Models table
+    def _load_model(self, graph):
+        # To store the database column names and their values to be inserted
+        columns = list()
+        values = list()
+
+        # Adding Graph attributes and their values to columns and values
+        attrs = vars(graph)
+        for item in attrs.items():
+            if (item[0] != 'nodes' and item[0] != 'start_node_indices' 
+                and item[0] != 'edges' and item[0] != 'adj_list'):
+                columns.append(item[0])
+                values.append(item[1])
+
+        columns = tuple(columns)
+        values = tuple(values)
+
+        # Inserting into database
+        with self.database.batch() as batch:
+            batch.insert(
+                table = 'Models',
+                columns = columns,
+                values = [values]
+            )
+
+            
+
+    # Internal function to load operator(Node) data into Operators table     
+    def _load_operators(self, graph):
+        # Surrogate Id for operators and iterating index
+        operator_id = 0
+
+        if len(graph.nodes) == 0:
+            return
+
+        # Number of nodes to be processed per batch 
+        num_attributes = len(vars(graph.nodes[0]))
+        num_nodes_per_batch = (20000 + num_attributes - 1) // num_attributes
+
+        num_nodes = len(graph.nodes)
+
+        # Inserting data into the Operators table
+
+        # Loop till all nodes are inserted
+        while operator_id < num_nodes:
+            # Creating and inserting a batch
```

Comment:
One thing to clarify here would be what would be semantics if after say k batches, you run out of space or there is some issue contacting Spanner and there are errors, and we still have remaining (n-k) batches pending. Would the graph be added to models table while the operators table will have incomplete set of entries? What would happen when we retry inserting the operators?

Reply:
Exception handling added in subsequent PR, in case of error, model is deleted from the database entirely and program terminates.

Reply:
Sounds good, thanks.